src/gui/TrackContentObjectView.cpp

```diff
@@ -1106,6 +1127,122 @@ void TrackContentObjectView::toggleMute( QVector<TrackContentObjectView *> tcovs
 	}
 }
 
+bool TrackContentObjectView::canMergeSelection(QVector<TrackContentObjectView*> tcovs)
+{
+	// We can only merge InstrumentTrack's TCOs, so check if we only have those in the selection,
+	// and also if they all belong to the same track
+
+	// We start assuming we can merge and if any condition is broken we set it to false
+	bool canMerge = true;
+
+	// Variable to check if all TCOs belong to the same track
+	TrackView* previousOwnerTrackView = nullptr;
+
+	// Then we check every selected TCO to see if all of them are InstrumentTrack's TCOs.
+	// If any isn't, we set canMerge to false and quit the loop.
+	for (auto tcov: tcovs)
+	{
+		TrackView* ownerTrackView = tcov->getTrackView();
+
+		// Set the previousOwnerTrackView to the first TrackView
+		if (!previousOwnerTrackView)
+		{
+			previousOwnerTrackView = ownerTrackView;
+		}
+
+		// If there are TCOs from different tracks or TCOs from tracks
+		// other than an InstrumentTrack, can't merge them
+		if (ownerTrackView != previousOwnerTrackView
+			|| !dynamic_cast<InstrumentTrackView*>(ownerTrackView))
```

Comment:
Instead of trying `dynamic_cast` for every `TrackView`s, you can check the type of the first `ownerTrackView` and then compare with the other `ownerTrackView`s without type checking.

Reply:
True! I'll move the `dynamic_cast` to the `if (!previousOwnerTrackView)` conditional and return false if it fails